pkg/cli/advisory_osv.go

```diff
@@ -3,42 +3,92 @@ package cli
-		Use:           "osv",
-		Short:         "Build an OSV dataset from Chainguard advisory data",
+		Use:   "osv",
+		Short: "Build an OSV dataset from Chainguard advisory data",
+		Long: `Build an OSV dataset from Chainguard advisory data.
+
+This command reads advisory data from one or more directories containing Chainguard
+advisory documents, and writes an OSV dataset to a local directory.
+
+Specify directories for advisory repositories using the --advisories-repo-dir flag.
+
+IMPORTANT: For now, the command assumes that the first listed advisory repository is the
+"Wolfi" repository, and that the rest are not. In the future, we might unify all advisory
+repositories into a single collection of all advisory documents, and remove the need for
+multiple advisory repositories.
+
+The user must also specify directories for all package repositories associated with the
+given advisory data. This is used to make sure the OSV data includes all relevant packages
+and subpackages.
+
+The output directory for the OSV dataset is specified using the --output flag. This
+directory must already exist before running the command.
+`,
 		SilenceErrors: true,
 		Args:          cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, _ []string) error {
 			ctx := cmd.Context()
 
 			if len(p.advisoriesRepoDirs) == 0 {
-				p.advisoriesRepoDirs = append(p.advisoriesRepoDirs, ".")
+				return fmt.Errorf("at least one advisory repository directory must be specified")
+			}
+
+			if len(p.packagesRepoDirs) == 0 {
+				return fmt.Errorf("at least one package repository directory must be specified")
+			}
+
+			if p.outputDirectory == "" {
+				return fmt.Errorf("output directory must be specified")
 			}
 
-			indices := make([]*configs.Index[v2.Document], 0, len(p.advisoriesRepoDirs))
+			advisoryIndices := make([]*configs.Index[v2.Document], 0, len(p.advisoriesRepoDirs))
 			for _, dir := range p.advisoriesRepoDirs {
-				advisoryFsys := rwos.DirFS(dir)
-				index, err := v2.NewIndex(cmd.Context(), advisoryFsys)
+				fsys := rwos.DirFS(dir)
+				index, err := v2.NewIndex(cmd.Context(), fsys)
 				if err != nil {
 					return fmt.Errorf("indexing advisory documents for directory %q: %w", dir, err)
 				}
 
-				indices = append(indices, index)
+				advisoryIndices = append(advisoryIndices, index)
+			}
+
+			packageIndices := make([]*configs.Index[config.Configuration], 0, len(p.packagesRepoDirs))
+			for _, dir := range p.packagesRepoDirs {
+				fsys := rwos.DirFS(dir)
+				index, err := build.NewIndex(cmd.Context(), fsys)
+				if err != nil {
+					return fmt.Errorf("indexing package build configurations for directory %q: %w", dir, err)
+				}
+
+				packageIndices = append(packageIndices, index)
+			}
+
+			addedEcosystems := []string{"Wolfi"}
+			for i := range p.advisoriesRepoDirs {
+				if i == 0 {
+					// Skip the first advisory repository, which is the "Wolfi" repository.
+					continue
+				}
+				addedEcosystems = append(addedEcosystems, "")
```

Comment:
what does empty string mean here?

Reply:
I kind of explain it here but I could do a better job — https://github.com/luhring/wolfictl/blob/5372e24fc044fc9787d0566809d415ff864d8605/pkg/advisory/osv.go#L35-L43

I'll update that. I'm hoping this is short-lived, we just need a way to sometimes add an extra ecosystem depending on which advisory dataset we're using.

Reply:
a2b5229